tests/test-sprite.php

```diff
@@ -0,0 +1,265 @@
+	 */
+	function test_with_dimensions() {
+
+		am_define_symbol(
+			[
+				'handle'    => 'with-dimensions',
+				'src'       => 'with-dimensions.svg',
+				'condition' => 'global',
+			]
+		);
+
+		$with_dimensions_expected = sprintf(
+			$this->empty_sprite_wrapper,
+			$this->clean_with_dimensions
+		);
+
+		$this->assertEquals(
+			$with_dimensions_expected,
+			\Asset_Manager_SVG_Sprite::instance()->sprite_document->C14N(),
+			'Should add the symbol to the sprite sheet.'
+		);
+
+		$basic_markup_expected = '<svg focusable="false" aria-hidden="true" width="24" height="24"><use href="#am-symbol-with-dimensions"></use></svg>';
+
+		$this->assertEquals(
+			$basic_markup_expected,
+			am_get_symbol( 'with-dimensions' ),
+			'Should get the svg + use markup.'
+		);
+
+		$this->assertArrayHasKey(
+			'focusable',
+			\Asset_Manager_SVG_Sprite::instance()->symbol_allowed_html['svg'],
+			'Should add global attributes to $symbol_allowed_html.'
+		);
+
+		$with_attributes_markup_expected = '<svg focusable="false" aria-hidden="true" width="48" height="48" class="am-test" data-test="test"><use href="#am-symbol-with-dimensions"></use></svg>';
+
+		$this->assertEquals(
+			$with_attributes_markup_expected,
+			am_get_symbol(
+				'with-dimensions',
+				[
+					'width'     => 48,
+					'class'     => 'am-test',
+					'data-test' => 'test'
+				]
+			),
+			'Should get the svg + use markup, with calculated height, global attributes, and additional attributes.'
+		);
+
+		$this->assertArrayHasKey(
+			'data-test',
+			\Asset_Manager_SVG_Sprite::instance()->symbol_allowed_html['svg'],
+			'Should add attributes to $symbol_allowed_html.'
+		);
+
+		$this->assertEquals(
+			$with_attributes_markup_expected,
+			get_echo(
+				'am_use_symbol',
+				[
+					'with-dimensions',
+					[
+						'width'     => 48,
+						'class'     => 'am-test',
+						'data-test' => 'test'
+					]
+				]
+			),
+			'Should echo the svg + use markup, with calculated height, global attributes, and additional attributes.'
+		);
+	}
+
+	/**
+	 * Test adding an asset with dimensions.
+	 */
+	function test_asset_with_export_junk() {
+
+		am_define_symbol(
+			[
+				'handle'    => 'export-junk',
+				'src'       => 'export-junk.svg',
+				'condition' => 'global',
+			]
+		);
+
+		$with_export_junk_expected = sprintf(
+			$this->empty_sprite_wrapper,
+			$this->with_export_junk
+		);
+
+		$this->assertEquals(
+			$with_export_junk_expected,
+			\Asset_Manager_SVG_Sprite::instance()->sprite_document->C14N(),
+			'Should add the symbol to the sprite sheet.'
+		);
+	}
+
+	/**
+	 * Test adding an asset with an embedded script tag.
+	 */
+	function test_asset_with_embedded_script() {
+
+		am_define_symbol(
+			[
+				'handle'    => 'embedded-script',
+				'src'       => 'danger.svg',
+				'condition' => 'global',
+			]
+		);
+
+		$with_embedded_script_expected = sprintf(
+			$this->empty_sprite_wrapper,
+			$this->with_embedded_script
```

Comment:
Should this be called `without_embedded_script` since this assumes the script is not embedded?